src/formatDate.js

```diff
@@ -8,7 +8,48 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let dateArr = [];
+  let resultString = '';
+  let index = 0;
+
+  if (date.includes('-')) {
+    dateArr = date.split('-');
+  } else if (date.includes('/')) {
+    dateArr = date.split('/');
+  } else {
+    dateArr = date.split('.');
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    const fromIndex = fromFormat.indexOf(toFormat[i]);
+
+    index++;
+
+    if (toFormat[i] === 'YY' && fromFormat.includes('YYYY')) {
+      const number = dateArr[index - 1];
+
+      resultString += number.slice(2);
+    } else if (toFormat[i] === 'YYYY' && fromFormat.includes('YY')) {
+      const year = Number(dateArr[index - 1]);
+
+      if (dateArr[index - 1] === '00') {
+        resultString += '2000.';
+        continue;
+      } else {
+        const fullYear = year < 30 ? `20${year}` : `19${year}`;
+
+        resultString += fullYear;
+      }
+    } else {
+      resultString += dateArr[fromIndex];
+    }
+
+    if (i < toFormat.length - 2) {
+      resultString += toFormat[3];
+    }
+  }
```

Comment:
The main loop of your function seems to be a bit complex and hard to read. It's also based on the assumption that the format of the date will always have 3 parts (year, month, day). This might not always be the case. Consider refactoring this part of the function to make it more readable and flexible.